app/controllers/api_controller/service_orders.rb

```diff
@@ -0,0 +1,13 @@
+class ApiController
+  module ServiceOrders
+    def clear_resource_service_orders(type, id, _data)
+      service_order = collection_class(type).find(id)
+      # service_order.clear
```

Comment:
let's add the begin/rescue on the api side for clear as it could raise an exception of shopping cart was already ordered.